src/wp-includes/functions.php

```diff
@@ -7635,3 +7633,72 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Wraps inline JavaScript in `<script>` tags.
+ *
+ * Enables to inject attributes in the `<script>` tag via the `script_attributes` filter.
+ * Automatically injects type attribute if needed.
+ *
+ * @param array     $attributes `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @param bool      $echo       Optional. Prints to the page if true, returns the value otherwise.
+ * @param string    $content    Optional. Code to be wrapped around `<script>` tags.
+ * @return string|null  Inline JavaScript code wrapped around <script> tags if $echo is true, null otherwise.
+ */
+function wp_script( $attributes, $double_quotes = true, $echo = true, $content = '' ) {
+	if ( ! array_key_exists( 'type', $attributes ) && ! is_admin() && ! current_theme_supports( 'html5', 'script' ) ) {
+		$attributes['type'] = 'text/javascript';
+	}
+
+	/**
+	 * Filters script attributes.
+	 *
+	 * @param string $attributes    Associative array representing script attributes.
+	 */
+	$attributes = apply_filters( 'script_attributes', $attributes );
```

Comment:
Hook names should be prefixed:
```suggestion
	$attributes = apply_filters( 'wp_script_attributes', $attributes );
```